This is the initial notebook to explore the data and run the first models. 

In [19]:
import pandas as pd
import numpy as np

from sklearn.impute import SimpleImputer
from sklearn.ensemble import RandomForestRegressor
import xgboost as xgb
from sklearn.preprocessing import OneHotEncoder

from fastai.imports import *
from fastai.tabular.all import *

from numpy import random
from sklearn.model_selection import train_test_split
from sklearn.utils import shuffle
from sklearn.metrics import mean_squared_error
from sklearn.metrics import mean_squared_log_error
from sklearn import metrics
from sklearn.model_selection import KFold, cross_val_score
from sklearn.model_selection import StratifiedKFold

import eli5
from eli5.sklearn import PermutationImportance

from scipy.stats import skew, norm
from catboost import CatBoostRegressor
from feature_engine.transformation import YeoJohnsonTransformer

import optuna
from optuna.samplers import TPESampler

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
test = pd.read_csv('test.csv')
sample = pd.read_csv('sample_submission.csv')
train = pd.read_csv('train.csv')

Based on initial analysis the train and test datasets have similar characteristics, so it will be easier to combine them for imputation and data analysis work. 

In [3]:
train_test = pd.concat([train, test], ignore_index=True)

In [4]:
pd.isnull(train_test).sum()[pd.isnull(train_test).sum() > 0]

MSZoning           4
LotFrontage      486
Alley           2721
Utilities          2
Exterior1st        1
Exterior2nd        1
MasVnrType        24
MasVnrArea        23
BsmtQual          81
BsmtCond          82
BsmtExposure      82
BsmtFinType1      79
BsmtFinSF1         1
BsmtFinType2      80
BsmtFinSF2         1
BsmtUnfSF          1
TotalBsmtSF        1
Electrical         1
BsmtFullBath       2
BsmtHalfBath       2
KitchenQual        1
Functional         2
FireplaceQu     1420
GarageType       157
GarageYrBlt      159
GarageFinish     159
GarageCars         1
GarageArea         1
GarageQual       159
GarageCond       159
PoolQC          2909
Fence           2348
MiscFeature     2814
SaleType           1
SalePrice       1459
dtype: int64

It looks like Alley, FireplaceQu, PoolQC, Fence and MiscFeature have significant numbers of missing data. So it will be best to eliminate those rows. There are a number of rows that have less than 5 rows with missing data. Since some of these are categorical and some are continuous data, their missing data will be replaced with the most frequent value. 

In [5]:
drop_high_nan=['Alley','FireplaceQu','PoolQC','Fence','MiscFeature']
train_test=train_test.drop(drop_high_nan,axis=1)
small_nan_cols = ['MSZoning', 'Utilities', 'Exterior1st', 'Exterior2nd', 'BsmtFinSF1', 'BsmtFinSF2', 'BsmtUnfSF', 
                  'TotalBsmtSF', 'Electrical', 'BsmtFullBath', 'BsmtHalfBath', 'KitchenQual', 'Functional', 'GarageCars', 
                  'GarageArea','SaleType', 'SaleCondition']
small_impute = SimpleImputer(strategy='most_frequent')
train_test[small_nan_cols] = pd.DataFrame(small_impute.fit_transform(train_test[small_nan_cols]),columns=small_nan_cols)

The following columns seem to have one value significantly larger than the rest, and it would probably be best to use the mode, or most common, value to feel each NaN value: MasVnrType, MasVnrArea, BsmtCond, BsmtExposure, BsmtFinType2, GarageType, GarageFinish, GarageQual, and GarageCond. That is represents 9 out of the 13 columns. 

BsmtQual has two values larger than the rest: Gd and TA. But it only has 2.8% NaNs, so simply using the mode might be good enough.  

GarageYrBlt has 59 NaNs out 2919 rows which is only 2%. It has a dispersed set of values, so it might be easiest just to have any NaNs have the same value as YearBuilt. 

BsmtFinType1 has only 2.7% value of NaNs, and most two of its largest values are GLQ and Unf. It might be easiest to use the mode here. 

LotFrontage has 486 NaNs out of 2919 rows which is a pretty high 16.7%. It has a dispersed range of values, but looking at its characteristics from the describe function above, it seems to have a pretty even distribution with a mean of 10,168 and a median of 9,453. So using the mean to fill in the NaNs seems reasonable. If it turns out this value has a high impact on the predictability of the SalePrice, then it might be good to revisit this assumption. 

In [6]:
mode_cols = ['MasVnrType', 'MasVnrArea', 'BsmtCond', 'BsmtExposure', 'BsmtFinType2', 'GarageType', 'GarageFinish', 
             'GarageQual','GarageCond', 'BsmtQual', 'BsmtFinType1']
mode_impute = SimpleImputer(strategy='most_frequent')
train_test[mode_cols] = pd.DataFrame(mode_impute.fit_transform(train_test[mode_cols]),columns=mode_cols)
train_test['LotFrontage'].fillna((train_test['LotFrontage'].mean()), inplace=True)
train_test['GarageYrBlt'] = train_test['GarageYrBlt'].fillna(train_test['YearBuilt'])

In [6]:
pd.isnull(train_test).sum()[pd.isnull(train_test).sum() > 0]

LotFrontage      486
MasVnrType        24
MasVnrArea        23
BsmtQual          81
BsmtCond          82
BsmtExposure      82
BsmtFinType1      79
BsmtFinType2      80
GarageType       157
GarageYrBlt      159
GarageFinish     159
GarageQual       159
GarageCond       159
SalePrice       1459
dtype: int64

In [7]:
train_test['BsmtQual'].unique()

array(['Gd', 'TA', 'Ex', nan, 'Fa'], dtype=object)

In [8]:
train_test['BsmtFinType1'].unique()

array(['GLQ', 'ALQ', 'Unf', 'Rec', 'BLQ', nan, 'LwQ'], dtype=object)

In [9]:
train_test.BsmtQual = train_test.BsmtQual.replace({"Ex": 110, "Gd": 95, "TA": 85, "Fa": 75, "Po": 60, "NA": 0})

In [10]:
train_test.BsmtFinType1 = train_test.BsmtFinType1.replace({"GLQ": 6, "ALQ": 5, "BLQ": 4, "Rec": 3, "LwQ": 2, "Unf": 1,
                                                         "NA": 0})

It will be necessary to identify all the columns that have non-numeric object values and then convert them to numeric values. 

In [11]:
obj_cols = list(train_test.select_dtypes(['object']).columns)
obj_cols

['MSZoning',
 'Street',
 'LotShape',
 'LandContour',
 'Utilities',
 'LotConfig',
 'LandSlope',
 'Neighborhood',
 'Condition1',
 'Condition2',
 'BldgType',
 'HouseStyle',
 'RoofStyle',
 'RoofMatl',
 'Exterior1st',
 'Exterior2nd',
 'MasVnrType',
 'ExterQual',
 'ExterCond',
 'Foundation',
 'BsmtCond',
 'BsmtExposure',
 'BsmtFinSF1',
 'BsmtFinType2',
 'BsmtFinSF2',
 'BsmtUnfSF',
 'TotalBsmtSF',
 'Heating',
 'HeatingQC',
 'CentralAir',
 'Electrical',
 'BsmtFullBath',
 'BsmtHalfBath',
 'KitchenQual',
 'Functional',
 'GarageType',
 'GarageFinish',
 'GarageCars',
 'GarageArea',
 'GarageQual',
 'GarageCond',
 'PavedDrive',
 'SaleType',
 'SaleCondition']

In [12]:
for column in obj_cols:
     train_test[column] = pd.factorize(train_test[column], sort=True)[0]

C:\Users\tompe\anaconda3\lib\site-packages\pandas\core\algorithms.py:798: FutureWarning: In a future version, the Index constructor will not infer numeric dtypes when passed object-dtype sequences (matching Series behavior)
  uniques = Index(uniques)


To create some new columns that might compound the effects of the existing columns with higher impact on the xgb score. 

In [13]:
train_test['QualCondSum'] = train_test['OverallQual'] + train_test['OverallCond']
train_test['RemodTime'] = train_test['YearRemodAdd'] - train_test['YearBuilt']
train_test['BsmtFinTypeSF1'] = train_test['BsmtFinType1'] * train_test['BsmtFinSF1']
train_test['TotalFlrSF'] = train_test['1stFlrSF'] + train_test['2ndFlrSF']
train_test['TotalFinSF'] = train_test['GrLivArea'] + train_test['BsmtFinSF1']
train_test['GarageCarArea'] = train_test['GarageArea'] * train_test['GarageCars']
train_test['TotalSF'] = train_test['1stFlrSF'] + train_test['2ndFlrSF'] + train_test['TotalBsmtSF']

To now eliminate the columns that have negative or zero influence on the xgb score. 

In [14]:
train_test.drop(['Utilities','Street','TotRmsAbvGrd','3SsnPorch','YrSold','Exterior2nd','LotConfig',
                              'HouseStyle','EnclosedPorch','WoodDeckSF','Foundation','RoofMatl','Electrical',
                              'GarageType','LotFrontage','SaleType','MoSold','BsmtExposure','1stFlrSF',
                              'BsmtFinSF1','Exterior1st','KitchenQual','TotalFlrSF'], axis=1, inplace=True)

To look at how many features have a skew above 0.6, since high skew can be an issue in regression analysis. 

In [15]:
skew_features = train_test.apply(lambda x: skew(x)).sort_values(ascending=False)

In [16]:
high_skew = skew_features[skew_features > 0.6]
skew_index = high_skew.index

In [17]:
skew_index

Index(['MiscVal', 'PoolArea', 'LotArea', 'LowQualFinSF', 'Heating',
       'Condition2', 'LandSlope', 'KitchenAbvGr', 'ScreenPorch',
       'BsmtHalfBath', 'BsmtFinSF2', 'Condition1', 'OpenPorchSF', 'BldgType',
       'RemodTime', 'RoofStyle', 'MSSubClass', 'GarageCarArea', 'GrLivArea',
       'TotalFinSF', 'TotalSF', '2ndFlrSF', 'Fireplaces', 'HalfBath',
       'BsmtFullBath'],
      dtype='object')

In [18]:
print("There are {} numerical features with Skew > 0.6 :".format(high_skew.shape[0]))
skewness = pd.DataFrame({'Skew' :high_skew})
skewness

There are 25 numerical features with Skew > 0.6 :


Skew
MiscVal        21.947195
PoolArea       16.898328
LotArea        12.822431
LowQualFinSF   12.088761
Heating        12.078788
Condition2     12.060093
LandSlope       4.975157
KitchenAbvGr    4.302254
ScreenPorch     3.946694
BsmtHalfBath    3.931594
BsmtFinSF2      3.476562
Condition1      2.983114
OpenPorchSF     2.535114
BldgType        2.192261
RemodTime       2.063712
RoofStyle       1.553307
MSSubClass      1.375457
GarageCarArea   1.296672
GrLivArea       1.269358
TotalFinSF      1.159630
TotalSF         0.922771
2ndFlrSF        0.861675
Fireplaces      0.733495
HalfBath        0.694566
BsmtFullBath    0.624832

To normalize the features with skew above 0.6.

In [20]:
yjt = YeoJohnsonTransformer()
yjt.fit(train_test[skew_index])
train_test[skew_index] = yjt.transform(train_test[skew_index])

In [22]:
skew_features = train_test.apply(lambda x: skew(x)).sort_values(ascending=False)
high_skew = skew_features[skew_features > 0.6]
skew_index = high_skew.index
skew_index

Index(['PoolArea', 'LowQualFinSF', 'MiscVal', 'LandSlope', 'BsmtHalfBath',
       'ScreenPorch', 'BsmtFinSF2', 'BldgType', 'Condition2'],
      dtype='object')

In [23]:
print("There are {} numerical features with Skew > 0.6 :".format(high_skew.shape[0]))
skewness = pd.DataFrame({'Skew' :high_skew})
skewness

There are 9 numerical features with Skew > 0.6 :


Skew
PoolArea      14.884318
LowQualFinSF   8.365937
MiscVal        5.037498
LandSlope      4.213417
BsmtHalfBath   3.707260
ScreenPorch    2.915215
BsmtFinSF2     2.356620
BldgType       1.765554
Condition2     1.384128

In [20]:
#for i in skew_index:
#    train_test[i] = np.log1p(train_test[i])

C:\Users\tompe\anaconda3\lib\site-packages\pandas\core\arraylike.py:397: RuntimeWarning: divide by zero encountered in log1p
  result = getattr(ufunc, method)(*inputs, **kwargs)


To create a column with the log of the SalePrice to match the evaluation process in the contest. 

In [24]:
train_test['LogSalePrice'] = train_test['SalePrice'].apply(np.log)

To separate the train_test dataset back into the train and test datasets, identify the independent and dependent columns, and create the validation split.

In [25]:
train = train_test[train_test['SalePrice'].notnull()].copy()
test = train_test[train_test['SalePrice'].isnull()].drop(['SalePrice','LogSalePrice'],axis=1)
X = train.drop(['SalePrice','LogSalePrice'],axis=1)
y = train.LogSalePrice

In [23]:
'''def xgb_objective(trial):
    
    param = {  
        #'tree_method':method,
        #'objective': 'reg:squaredlogerror',
        'eval_metric': 'rmse',
        'lambda': trial.suggest_float('lambda', 0, 10.0),
        'alpha': trial.suggest_float('alpha', 0, 10.0),
        'colsample_bytree': trial.suggest_float('colsample_bytree', 0.1,1.0),
        'subsample': trial.suggest_float('subsample', 0.2,1.0),
        'learning_rate': trial.suggest_float('learning_rate', 0.0001,0.1),
        'n_estimators': trial.suggest_int('n_estimators', 100,3000),
        'max_depth': trial.suggest_categorical('max_depth', [2,3,4,5,6,7,8,9,10]),
        'min_child_weight': trial.suggest_int('min_child_weight', 1, 10),
        'num_parallel_tree': trial.suggest_int('num_parallel_tree',1,1),
        }

    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)
    model = xgb.XGBRegressor(**param)
    model.fit(X_train, y_train, eval_set=[(X_test, y_test)], verbose=0)
    y_pred = model.predict(X_test)
    rmse = np.sqrt(mean_squared_error(y_test, y_pred))
    return rmse'''

In [24]:
'''study_xgb = optuna.create_study(sampler=TPESampler(), direction='minimize')
study_xgb.optimize(xgb_objective, n_trials=1000)
print('Number of finished trials:', len(study_xgb.trials))
print('Best trial:', study_xgb.best_trial.params)'''

[I 2023-06-06 07:55:24,660] A new study created in memory with name: no-name-8e48ddf0-2766-4099-bc18-c57b6583c40c
[I 2023-06-06 07:55:27,892] Trial 0 finished with value: 0.15449259040773541 and parameters: {'lambda': 5.005512412442891, 'alpha': 7.507617123368103, 'colsample_bytree': 0.5172618900102177, 'subsample': 0.28895669552474684, 'learning_rate': 0.06955157564922773, 'n_estimators': 2231, 'max_depth': 9, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 0 with value: 0.15449259040773541.
[I 2023-06-06 07:55:31,762] Trial 1 finished with value: 0.13641266132782368 and parameters: {'lambda': 9.712140075283925, 'alpha': 6.987322929271834, 'colsample_bytree': 0.9948400112674631, 'subsample': 0.9455138711490416, 'learning_rate': 0.01663248887679928, 'n_estimators': 2453, 'max_depth': 3, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 1 with value: 0.13641266132782368.
[I 2023-06-06 07:55:34,273] Trial 2 finished with value: 0.15202119138544057 and parameters

[I 2023-06-06 07:56:18,066] Trial 20 finished with value: 0.12514377783784233 and parameters: {'lambda': 2.180374322134986, 'alpha': 2.0955928782841804, 'colsample_bytree': 0.2047772530002056, 'subsample': 0.5610717025710661, 'learning_rate': 0.023349243814958648, 'n_estimators': 1541, 'max_depth': 2, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 19 with value: 0.11033085615589797.
[I 2023-06-06 07:56:21,053] Trial 21 finished with value: 0.1090806721241984 and parameters: {'lambda': 1.2873257231394375, 'alpha': 1.0302040674616142, 'colsample_bytree': 0.444424840534136, 'subsample': 0.5632870298355229, 'learning_rate': 0.025907671991715484, 'n_estimators': 1123, 'max_depth': 6, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 21 with value: 0.1090806721241984.
[I 2023-06-06 07:56:23,683] Trial 22 finished with value: 0.14973650991920073 and parameters: {'lambda': 1.1375735776627849, 'alpha': 1.3042580152021617, 'colsample_bytree': 0.42681287965974557, 'subs

[I 2023-06-06 07:57:29,846] Trial 40 finished with value: 0.1525342887620296 and parameters: {'lambda': 4.677566585686249, 'alpha': 0.7569829006274198, 'colsample_bytree': 0.4855644178502802, 'subsample': 0.716862543107563, 'learning_rate': 0.03706059578951099, 'n_estimators': 370, 'max_depth': 3, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 21 with value: 0.1090806721241984.
[I 2023-06-06 07:57:34,191] Trial 41 finished with value: 0.1353370494923925 and parameters: {'lambda': 2.909699801648847, 'alpha': 2.0296881704211196, 'colsample_bytree': 0.38398860438615784, 'subsample': 0.5295630710273964, 'learning_rate': 0.0382498805531267, 'n_estimators': 1299, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 21 with value: 0.1090806721241984.
[I 2023-06-06 07:57:39,268] Trial 42 finished with value: 0.11411518796798986 and parameters: {'lambda': 2.260384467778971, 'alpha': 1.7430613247745856, 'colsample_bytree': 0.38743059856151657, 'subsample':

[I 2023-06-06 07:59:35,124] Trial 60 finished with value: 0.11358476826674026 and parameters: {'lambda': 1.435115948690739, 'alpha': 3.7587311553239195, 'colsample_bytree': 0.37060369425378464, 'subsample': 0.5800483024681101, 'learning_rate': 0.01809680041270406, 'n_estimators': 2374, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 44 with value: 0.10740699373958382.
[I 2023-06-06 07:59:41,099] Trial 61 finished with value: 0.13367274688921862 and parameters: {'lambda': 1.4275710393178218, 'alpha': 3.258011664012228, 'colsample_bytree': 0.36536722347656975, 'subsample': 0.5749007962728189, 'learning_rate': 0.018873688894383508, 'n_estimators': 2341, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 44 with value: 0.10740699373958382.
[I 2023-06-06 07:59:46,404] Trial 62 finished with value: 0.14407559245147164 and parameters: {'lambda': 2.028696809275548, 'alpha': 3.7906406046435586, 'colsample_bytree': 0.403982528487307, 'subs

[I 2023-06-06 08:01:10,998] Trial 80 finished with value: 0.14248380659608137 and parameters: {'lambda': 3.1724023780981785, 'alpha': 2.6370522951517406, 'colsample_bytree': 0.4871320344324041, 'subsample': 0.46202624117017144, 'learning_rate': 0.028498599482763183, 'n_estimators': 829, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 44 with value: 0.10740699373958382.
[I 2023-06-06 08:01:16,074] Trial 81 finished with value: 0.12674532397045568 and parameters: {'lambda': 2.193727951423217, 'alpha': 1.7068752487716092, 'colsample_bytree': 0.38740186511577523, 'subsample': 0.5471548737486193, 'learning_rate': 0.043873922436891984, 'n_estimators': 1709, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 44 with value: 0.10740699373958382.
[I 2023-06-06 08:01:21,045] Trial 82 finished with value: 0.13387186400956244 and parameters: {'lambda': 3.7466592850481506, 'alpha': 2.1765597504926513, 'colsample_bytree': 0.43093752209245223, '

[I 2023-06-06 08:02:44,305] Trial 100 finished with value: 0.13953356778427775 and parameters: {'lambda': 2.3283627867593113, 'alpha': 1.0397959288779193, 'colsample_bytree': 0.2746193867030984, 'subsample': 0.5624582465203475, 'learning_rate': 0.04108929574977398, 'n_estimators': 1443, 'max_depth': 9, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:02:50,669] Trial 101 finished with value: 0.11073203943615031 and parameters: {'lambda': 1.1897429585054509, 'alpha': 2.783154414182268, 'colsample_bytree': 0.36878620225782655, 'subsample': 0.5994027605528239, 'learning_rate': 0.020449213548542955, 'n_estimators': 2356, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:02:54,567] Trial 102 finished with value: 0.12565078998312 and parameters: {'lambda': 1.166934858524403, 'alpha': 2.2749515887345546, 'colsample_bytree': 0.3684283582553695, 'sub

[I 2023-06-06 08:04:03,510] Trial 120 finished with value: 0.12223492460143191 and parameters: {'lambda': 2.1341417211035623, 'alpha': 1.3200901267390743, 'colsample_bytree': 0.2807363794325548, 'subsample': 0.5949670170380826, 'learning_rate': 0.020469963916770424, 'n_estimators': 1029, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:04:10,296] Trial 121 finished with value: 0.12896142017069676 and parameters: {'lambda': 2.0615052312348863, 'alpha': 2.9713328640618486, 'colsample_bytree': 0.3903181373323059, 'subsample': 0.5941280582966039, 'learning_rate': 0.023915145557837563, 'n_estimators': 2294, 'max_depth': 7, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:04:16,480] Trial 122 finished with value: 0.13410958994108543 and parameters: {'lambda': 1.703807873849561, 'alpha': 2.9174783660025, 'colsample_bytree': 0.409897432959347, 'sub

[I 2023-06-06 08:05:55,893] Trial 140 finished with value: 0.1253863826130969 and parameters: {'lambda': 0.7478932415746129, 'alpha': 2.8136885538762404, 'colsample_bytree': 0.3868381020886081, 'subsample': 0.501069805358808, 'learning_rate': 0.01881997125078241, 'n_estimators': 2402, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:06:00,776] Trial 141 finished with value: 0.10857954764173677 and parameters: {'lambda': 2.8045093759027684, 'alpha': 1.1337227038564517, 'colsample_bytree': 0.352260488193728, 'subsample': 0.6012943160266213, 'learning_rate': 0.03154862732852564, 'n_estimators': 1549, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:06:05,923] Trial 142 finished with value: 0.12131389409422172 and parameters: {'lambda': 2.4424786415598057, 'alpha': 0.6849441599067415, 'colsample_bytree': 0.3488984052488636, 'sub

[I 2023-06-06 08:07:32,687] Trial 160 finished with value: 0.12073716580492401 and parameters: {'lambda': 2.799573179061431, 'alpha': 0.3997206093381164, 'colsample_bytree': 0.34910824852507377, 'subsample': 0.5132506294070406, 'learning_rate': 0.03259719034159043, 'n_estimators': 2116, 'max_depth': 4, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:07:38,501] Trial 161 finished with value: 0.12770589075003655 and parameters: {'lambda': 1.527734829941867, 'alpha': 3.168491867059466, 'colsample_bytree': 0.3880133988681714, 'subsample': 0.5798618730509659, 'learning_rate': 0.017618226057462764, 'n_estimators': 2323, 'max_depth': 5, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 85 with value: 0.10394775010426308.
[I 2023-06-06 08:07:43,132] Trial 162 finished with value: 0.1357862415397399 and parameters: {'lambda': 1.9255014466479232, 'alpha': 2.6457901479214194, 'colsample_bytree': 0.3679837192428051, 'su

[I 2023-06-06 08:08:49,602] Trial 180 finished with value: 0.11659127130676189 and parameters: {'lambda': 2.582422917444424, 'alpha': 0.6383285958824687, 'colsample_bytree': 0.44413929403240393, 'subsample': 0.6290149621923924, 'learning_rate': 0.03107040224893135, 'n_estimators': 1304, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:08:52,235] Trial 181 finished with value: 0.15023591957448204 and parameters: {'lambda': 1.183905532836887, 'alpha': 3.2301784856291995, 'colsample_bytree': 0.3829404382524277, 'subsample': 0.5814074577722202, 'learning_rate': 0.022671577729114652, 'n_estimators': 1146, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:08:57,933] Trial 182 finished with value: 0.15151156682278383 and parameters: {'lambda': 1.3845638198801162, 'alpha': 4.314050176272422, 'colsample_bytree': 0.36820855585175927,

[I 2023-06-06 08:10:14,266] Trial 200 finished with value: 0.11376476360346892 and parameters: {'lambda': 2.1700921350047624, 'alpha': 1.3309144454598423, 'colsample_bytree': 0.3855641203439489, 'subsample': 0.5719912147976155, 'learning_rate': 0.0465712999574239, 'n_estimators': 2259, 'max_depth': 6, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:10:19,447] Trial 201 finished with value: 0.13230617429197394 and parameters: {'lambda': 2.4706742810110986, 'alpha': 1.1615432411622475, 'colsample_bytree': 0.3443000563820661, 'subsample': 0.609272640439846, 'learning_rate': 0.03165824897452948, 'n_estimators': 1592, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:10:23,725] Trial 202 finished with value: 0.12083703490547333 and parameters: {'lambda': 2.3336277086656247, 'alpha': 0.736503537035712, 'colsample_bytree': 0.3275339499255925, 's

[I 2023-06-06 08:12:07,357] Trial 220 finished with value: 0.14924363131466445 and parameters: {'lambda': 2.583482637119186, 'alpha': 0.33387866751398, 'colsample_bytree': 0.4845748421217381, 'subsample': 0.515219062346379, 'learning_rate': 0.03871661555140632, 'n_estimators': 1918, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:12:12,413] Trial 221 finished with value: 0.1262788558905411 and parameters: {'lambda': 1.8102506094606152, 'alpha': 0.6852421069125796, 'colsample_bytree': 0.5058567645460141, 'subsample': 0.5776263164105289, 'learning_rate': 0.03575430818032862, 'n_estimators': 2029, 'max_depth': 2, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:12:17,281] Trial 222 finished with value: 0.12993138701328627 and parameters: {'lambda': 2.1270666728216456, 'alpha': 3.5238413486719455, 'colsample_bytree': 0.4377361599287906, 'sub

[I 2023-06-06 08:13:38,998] Trial 240 finished with value: 0.11913044191340375 and parameters: {'lambda': 3.054309868774656, 'alpha': 1.1105874262073034, 'colsample_bytree': 0.3061604006924571, 'subsample': 0.6293356553788296, 'learning_rate': 0.03952043971436143, 'n_estimators': 1673, 'max_depth': 6, 'min_child_weight': 3, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:13:43,619] Trial 241 finished with value: 0.11981417088464331 and parameters: {'lambda': 2.882138474490413, 'alpha': 1.4090932160235652, 'colsample_bytree': 0.34823980721186815, 'subsample': 0.6062632403726664, 'learning_rate': 0.035268655786837466, 'n_estimators': 1701, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:13:48,535] Trial 242 finished with value: 0.12200605483123769 and parameters: {'lambda': 3.9152186308240604, 'alpha': 0.7518067623110843, 'colsample_bytree': 0.38519988541880046

[I 2023-06-06 08:15:16,539] Trial 260 finished with value: 0.11580598117843666 and parameters: {'lambda': 2.7792114129261605, 'alpha': 0.6989675894387518, 'colsample_bytree': 0.355439416604463, 'subsample': 0.5892983860184478, 'learning_rate': 0.07568624547212038, 'n_estimators': 1486, 'max_depth': 10, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:15:22,144] Trial 261 finished with value: 0.12769258200155284 and parameters: {'lambda': 2.4675685624674317, 'alpha': 0.004166561632103605, 'colsample_bytree': 0.4116348185657055, 'subsample': 0.5206582969710437, 'learning_rate': 0.03602426269706284, 'n_estimators': 1565, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:15:25,901] Trial 262 finished with value: 0.12595152309778102 and parameters: {'lambda': 2.195428878055551, 'alpha': 2.2123668263103107, 'colsample_bytree': 0.8185298655160584

[I 2023-06-06 08:16:36,740] Trial 280 finished with value: 0.12140426826793668 and parameters: {'lambda': 2.4669637889695433, 'alpha': 3.091313583410777, 'colsample_bytree': 0.39279800269816806, 'subsample': 0.5396405498387993, 'learning_rate': 0.027293369188854208, 'n_estimators': 1586, 'max_depth': 5, 'min_child_weight': 2, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:16:41,094] Trial 281 finished with value: 0.14052444484677673 and parameters: {'lambda': 2.2037837963625173, 'alpha': 3.2934035095727134, 'colsample_bytree': 0.36729179998452377, 'subsample': 0.5832036948843914, 'learning_rate': 0.03205829833242273, 'n_estimators': 1525, 'max_depth': 6, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:16:46,088] Trial 282 finished with value: 0.13046653059432775 and parameters: {'lambda': 2.9672286703537054, 'alpha': 2.7340746556001383, 'colsample_bytree': 0.274704105792819

[I 2023-06-06 08:17:58,529] Trial 300 finished with value: 0.12503235990815423 and parameters: {'lambda': 4.133319880033939, 'alpha': 2.271009738411898, 'colsample_bytree': 0.45301022185666184, 'subsample': 0.7333051166305934, 'learning_rate': 0.01629540434253083, 'n_estimators': 1028, 'max_depth': 9, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:18:01,160] Trial 301 finished with value: 0.13796232578155834 and parameters: {'lambda': 4.118277751272671, 'alpha': 2.59383493503335, 'colsample_bytree': 0.297997166349747, 'subsample': 0.748675023689138, 'learning_rate': 0.02078344447167713, 'n_estimators': 895, 'max_depth': 9, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:18:04,121] Trial 302 finished with value: 0.12226779352622182 and parameters: {'lambda': 1.8634710704834145, 'alpha': 2.823588583527451, 'colsample_bytree': 0.387261587960303, 'subsamp

[I 2023-06-06 08:19:15,817] Trial 320 finished with value: 0.12491694035669698 and parameters: {'lambda': 4.564912212702799, 'alpha': 1.3674533372485929, 'colsample_bytree': 0.39115145310019594, 'subsample': 0.7611331733056556, 'learning_rate': 0.07186238169990611, 'n_estimators': 1128, 'max_depth': 10, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:19:19,661] Trial 321 finished with value: 0.12487900134366668 and parameters: {'lambda': 1.7929193397431376, 'alpha': 1.2177594161291287, 'colsample_bytree': 0.3710867199555946, 'subsample': 0.6251186533533711, 'learning_rate': 0.06361471378694897, 'n_estimators': 1206, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:19:22,593] Trial 322 finished with value: 0.1370387458525613 and parameters: {'lambda': 4.492073467390426, 'alpha': 1.6058984272471029, 'colsample_bytree': 0.4030885725800982, 

[I 2023-06-06 08:20:24,249] Trial 340 finished with value: 0.1333443720444566 and parameters: {'lambda': 2.9392721418085053, 'alpha': 3.186792536642078, 'colsample_bytree': 0.33112478954315366, 'subsample': 0.4807761162646593, 'learning_rate': 0.07589049255008996, 'n_estimators': 1273, 'max_depth': 2, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:20:27,115] Trial 341 finished with value: 0.1308872281230049 and parameters: {'lambda': 0.6549459274833788, 'alpha': 2.4905172324721336, 'colsample_bytree': 0.2990257981949679, 'subsample': 0.5256968185727889, 'learning_rate': 0.02325133363125539, 'n_estimators': 928, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:20:34,408] Trial 342 finished with value: 0.13870244589162126 and parameters: {'lambda': 2.6595401036595163, 'alpha': 2.9753820339130277, 'colsample_bytree': 0.38338458976594486, '

[I 2023-06-06 08:21:44,588] Trial 360 finished with value: 0.11874847294840078 and parameters: {'lambda': 2.496400927942352, 'alpha': 3.177401717159439, 'colsample_bytree': 0.4748392221281564, 'subsample': 0.5792619438199818, 'learning_rate': 0.045801350892240335, 'n_estimators': 1650, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:21:49,141] Trial 361 finished with value: 0.1274925044585784 and parameters: {'lambda': 4.303243163631731, 'alpha': 2.6441128063820534, 'colsample_bytree': 0.32853597836751786, 'subsample': 0.5316984728049798, 'learning_rate': 0.06148040370172855, 'n_estimators': 1785, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:21:52,079] Trial 362 finished with value: 0.10838545194902405 and parameters: {'lambda': 2.195733970559657, 'alpha': 0.9980364238169557, 'colsample_bytree': 0.3997730423979304, 's

[I 2023-06-06 08:23:08,294] Trial 380 finished with value: 0.12679711600540672 and parameters: {'lambda': 2.4329866181856965, 'alpha': 1.5460344567922584, 'colsample_bytree': 0.3729231025238976, 'subsample': 0.8024265250384129, 'learning_rate': 0.05107810637925528, 'n_estimators': 1284, 'max_depth': 2, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:23:16,723] Trial 381 finished with value: 0.12386944936507262 and parameters: {'lambda': 1.8887934859477777, 'alpha': 1.307959268642552, 'colsample_bytree': 0.39486493244863247, 'subsample': 0.5832663550876556, 'learning_rate': 0.031562952769683045, 'n_estimators': 2872, 'max_depth': 6, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:23:22,439] Trial 382 finished with value: 0.11103842068891423 and parameters: {'lambda': 5.47625835023324, 'alpha': 0.8877471406784994, 'colsample_bytree': 0.35694066789620477,

[I 2023-06-06 08:24:34,983] Trial 400 finished with value: 0.11862547605633297 and parameters: {'lambda': 3.7394605540503068, 'alpha': 0.6359938378511666, 'colsample_bytree': 0.9216709583253304, 'subsample': 0.5292244118161124, 'learning_rate': 0.06807840951460167, 'n_estimators': 1057, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:24:38,559] Trial 401 finished with value: 0.12274043742872934 and parameters: {'lambda': 2.1042372644256133, 'alpha': 1.559337482417734, 'colsample_bytree': 0.35279925491074215, 'subsample': 0.47869083409331814, 'learning_rate': 0.04983585578572523, 'n_estimators': 1456, 'max_depth': 2, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:24:42,230] Trial 402 finished with value: 0.13744008813321051 and parameters: {'lambda': 1.4204709512538025, 'alpha': 1.325121455351086, 'colsample_bytree': 0.31190826601741634

[I 2023-06-06 08:25:49,201] Trial 420 finished with value: 0.14322534741460732 and parameters: {'lambda': 1.3409022810569173, 'alpha': 2.8284315637314723, 'colsample_bytree': 0.3346663472852243, 'subsample': 0.6668079982038704, 'learning_rate': 0.039737974335751866, 'n_estimators': 872, 'max_depth': 10, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:25:53,349] Trial 421 finished with value: 0.14259243121676723 and parameters: {'lambda': 3.2665649469487796, 'alpha': 6.8010981909863295, 'colsample_bytree': 0.29885167968146464, 'subsample': 0.5219240114521375, 'learning_rate': 0.0985254701327198, 'n_estimators': 1698, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:25:59,191] Trial 422 finished with value: 0.1267940643462233 and parameters: {'lambda': 4.74161087661067, 'alpha': 1.2717166892839133, 'colsample_bytree': 0.43433389380304793, 

[I 2023-06-06 08:27:14,352] Trial 440 finished with value: 0.12428256563641873 and parameters: {'lambda': 6.966123572346264, 'alpha': 0.3370830813536858, 'colsample_bytree': 0.7449134636320238, 'subsample': 0.6133251083192217, 'learning_rate': 0.023379448711453442, 'n_estimators': 1512, 'max_depth': 7, 'min_child_weight': 3, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:27:17,941] Trial 441 finished with value: 0.15816136854327525 and parameters: {'lambda': 3.3388473458250685, 'alpha': 4.954542561278939, 'colsample_bytree': 0.26592688473289194, 'subsample': 0.683914021460144, 'learning_rate': 0.02058291763132344, 'n_estimators': 1364, 'max_depth': 8, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:27:22,319] Trial 442 finished with value: 0.1401382213371716 and parameters: {'lambda': 5.827695849467535, 'alpha': 2.34269220297041, 'colsample_bytree': 0.5349964104446077, 'sub

[I 2023-06-06 08:28:43,682] Trial 460 finished with value: 0.14647509941178521 and parameters: {'lambda': 1.2158128011684586, 'alpha': 1.8444552739178373, 'colsample_bytree': 0.38488986658883984, 'subsample': 0.7012964989984288, 'learning_rate': 0.019053624475110872, 'n_estimators': 1183, 'max_depth': 6, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:28:46,659] Trial 461 finished with value: 0.12509287575947506 and parameters: {'lambda': 6.00792101846482, 'alpha': 0.24600163211563897, 'colsample_bytree': 0.4107043952581698, 'subsample': 0.31890098769820746, 'learning_rate': 0.007234346448372758, 'n_estimators': 1035, 'max_depth': 6, 'min_child_weight': 2, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:28:49,496] Trial 462 finished with value: 0.14500098783912135 and parameters: {'lambda': 6.357265168229399, 'alpha': 4.09842394359901, 'colsample_bytree': 0.44006831116715905

[I 2023-06-06 08:30:13,054] Trial 480 finished with value: 0.14568123719579792 and parameters: {'lambda': 2.7457057247202172, 'alpha': 5.986101202849498, 'colsample_bytree': 0.942695321848071, 'subsample': 0.8187251926791996, 'learning_rate': 0.018889882276499584, 'n_estimators': 1508, 'max_depth': 4, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:30:17,683] Trial 481 finished with value: 0.11170451595449707 and parameters: {'lambda': 2.608714495505668, 'alpha': 1.3742920376591103, 'colsample_bytree': 0.5836458974450904, 'subsample': 0.9636696431402383, 'learning_rate': 0.022186822193789722, 'n_estimators': 1615, 'max_depth': 5, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:30:22,564] Trial 482 finished with value: 0.13660403182122338 and parameters: {'lambda': 2.9378987934288365, 'alpha': 2.262430958674011, 'colsample_bytree': 0.5971403493028671, '

[I 2023-06-06 08:31:42,684] Trial 500 finished with value: 0.11595644920929055 and parameters: {'lambda': 3.2601228726286307, 'alpha': 2.977323266501802, 'colsample_bytree': 0.5820724414999946, 'subsample': 0.9023066502449707, 'learning_rate': 0.034438827683686335, 'n_estimators': 1559, 'max_depth': 5, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:31:45,284] Trial 501 finished with value: 0.12985179351640316 and parameters: {'lambda': 5.433869076888252, 'alpha': 3.739458909283548, 'colsample_bytree': 0.631690400795897, 'subsample': 0.8291539029110516, 'learning_rate': 0.027339801260788756, 'n_estimators': 941, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:31:48,655] Trial 502 finished with value: 0.12125740647556098 and parameters: {'lambda': 4.730668470810054, 'alpha': 3.468855014064389, 'colsample_bytree': 0.30412068259366487, 'su

[I 2023-06-06 08:33:02,937] Trial 520 finished with value: 0.1364473918428143 and parameters: {'lambda': 9.458867222682263, 'alpha': 2.7036842599232824, 'colsample_bytree': 0.6451422132395663, 'subsample': 0.845141592086603, 'learning_rate': 0.038255079851673174, 'n_estimators': 1471, 'max_depth': 7, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:33:07,978] Trial 521 finished with value: 0.12346584870255972 and parameters: {'lambda': 8.558114480824308, 'alpha': 2.538371058665257, 'colsample_bytree': 0.608500203752252, 'subsample': 0.8051788669044406, 'learning_rate': 0.04041327585263021, 'n_estimators': 1564, 'max_depth': 10, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:33:11,178] Trial 522 finished with value: 0.12876842843808511 and parameters: {'lambda': 7.3698652119593895, 'alpha': 2.890581167963583, 'colsample_bytree': 0.6262950359666843, 'sub

[I 2023-06-06 08:34:22,173] Trial 540 finished with value: 0.16878151301784222 and parameters: {'lambda': 8.76253381102727, 'alpha': 7.6115536903517915, 'colsample_bytree': 0.1292658243943085, 'subsample': 0.5329913394284521, 'learning_rate': 0.03722082694013275, 'n_estimators': 1617, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:34:26,085] Trial 541 finished with value: 0.1342952507521942 and parameters: {'lambda': 9.025512999628251, 'alpha': 5.496017376208325, 'colsample_bytree': 0.33067105065259816, 'subsample': 0.7327111978815672, 'learning_rate': 0.022509763843397597, 'n_estimators': 1512, 'max_depth': 10, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:34:28,904] Trial 542 finished with value: 0.14991972855375305 and parameters: {'lambda': 8.409017863142045, 'alpha': 0.9744505082405808, 'colsample_bytree': 0.378044718670864, 'su

[I 2023-06-06 08:35:44,325] Trial 560 finished with value: 0.11949398965267317 and parameters: {'lambda': 8.70573340147841, 'alpha': 0.6895449665487987, 'colsample_bytree': 0.3891278347878338, 'subsample': 0.7168381117676123, 'learning_rate': 0.0385360796257671, 'n_estimators': 1534, 'max_depth': 5, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:35:49,176] Trial 561 finished with value: 0.11391084218234804 and parameters: {'lambda': 7.478429289542365, 'alpha': 1.87092857623457, 'colsample_bytree': 0.4071475293767512, 'subsample': 0.4722340363541868, 'learning_rate': 0.022830828387142486, 'n_estimators': 1632, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:35:51,571] Trial 562 finished with value: 0.13802913575756626 and parameters: {'lambda': 9.918275761783686, 'alpha': 3.0338888754583886, 'colsample_bytree': 0.34591737329189604, 'sub

[I 2023-06-06 08:36:55,506] Trial 580 finished with value: 0.11102736007760171 and parameters: {'lambda': 7.60711349821039, 'alpha': 1.319068395876048, 'colsample_bytree': 0.26363757556850426, 'subsample': 0.7370949404877299, 'learning_rate': 0.04334062597792467, 'n_estimators': 1024, 'max_depth': 5, 'min_child_weight': 3, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:36:59,000] Trial 581 finished with value: 0.1204097700775472 and parameters: {'lambda': 3.847409579228487, 'alpha': 3.9026859093655784, 'colsample_bytree': 0.23205652811104033, 'subsample': 0.5371462235837531, 'learning_rate': 0.039115907153970696, 'n_estimators': 1256, 'max_depth': 5, 'min_child_weight': 2, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:37:02,355] Trial 582 finished with value: 0.12370071280791503 and parameters: {'lambda': 7.939772112671433, 'alpha': 1.606629761861549, 'colsample_bytree': 0.27274131317140926, 's

[I 2023-06-06 08:38:14,217] Trial 600 finished with value: 0.13231262982938818 and parameters: {'lambda': 4.029621666923085, 'alpha': 1.252494295221771, 'colsample_bytree': 0.5709116012468122, 'subsample': 0.5277236932415754, 'learning_rate': 0.03584519735840477, 'n_estimators': 1459, 'max_depth': 2, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:38:21,734] Trial 601 finished with value: 0.13524423600540772 and parameters: {'lambda': 1.6602314400745, 'alpha': 1.7274524697540912, 'colsample_bytree': 0.20012411983693656, 'subsample': 0.7474397617041368, 'learning_rate': 0.06906682621003316, 'n_estimators': 1942, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:38:26,092] Trial 602 finished with value: 0.1305723402795281 and parameters: {'lambda': 1.3679804021661974, 'alpha': 1.570309608575214, 'colsample_bytree': 0.2659251550918828, 'sub

[I 2023-06-06 08:39:57,361] Trial 620 finished with value: 0.12330785961245824 and parameters: {'lambda': 3.649601302547172, 'alpha': 2.5961366122164344, 'colsample_bytree': 0.37559334756563933, 'subsample': 0.5733834133633375, 'learning_rate': 0.03144622497377564, 'n_estimators': 1806, 'max_depth': 5, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:40:03,215] Trial 621 finished with value: 0.10776503919175431 and parameters: {'lambda': 2.824829083110039, 'alpha': 2.0177702919584166, 'colsample_bytree': 0.2581136241907523, 'subsample': 0.8079577278475494, 'learning_rate': 0.01633219305590458, 'n_estimators': 2156, 'max_depth': 3, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:40:08,815] Trial 622 finished with value: 0.14919296576517282 and parameters: {'lambda': 2.8176619401357423, 'alpha': 5.168406709146273, 'colsample_bytree': 0.20472293406960546, 

[I 2023-06-06 08:41:49,055] Trial 640 finished with value: 0.11832621416909254 and parameters: {'lambda': 3.0468785565089487, 'alpha': 0.6119927156034048, 'colsample_bytree': 0.24552811920073264, 'subsample': 0.820599064799659, 'learning_rate': 0.09579837080846429, 'n_estimators': 842, 'max_depth': 2, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:41:57,501] Trial 641 finished with value: 0.12420116066003509 and parameters: {'lambda': 0.7670211540482224, 'alpha': 1.343952115992796, 'colsample_bytree': 0.26955011489284797, 'subsample': 0.7615964996454931, 'learning_rate': 0.031248896461345578, 'n_estimators': 1860, 'max_depth': 9, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:42:03,049] Trial 642 finished with value: 0.11849708299125027 and parameters: {'lambda': 5.725859098390682, 'alpha': 4.27311241005833, 'colsample_bytree': 0.3545944501264331, 's

[I 2023-06-06 08:43:58,933] Trial 660 finished with value: 0.13154566271406892 and parameters: {'lambda': 9.437267179567574, 'alpha': 1.008643895911146, 'colsample_bytree': 0.43443938927433534, 'subsample': 0.587110708787242, 'learning_rate': 0.09325713222128301, 'n_estimators': 884, 'max_depth': 6, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:44:04,920] Trial 661 finished with value: 0.1256661781109034 and parameters: {'lambda': 4.020480559992076, 'alpha': 3.657967217184533, 'colsample_bytree': 0.39267519861880296, 'subsample': 0.6770200122292118, 'learning_rate': 0.054271654990030706, 'n_estimators': 1507, 'max_depth': 7, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:44:09,638] Trial 662 finished with value: 0.1378596554076323 and parameters: {'lambda': 1.196873200520046, 'alpha': 3.4001630875778424, 'colsample_bytree': 0.8894914996139855, 'subs

[I 2023-06-06 08:46:01,739] Trial 680 finished with value: 0.11369797775980178 and parameters: {'lambda': 7.608030916660112, 'alpha': 0.5520876420676077, 'colsample_bytree': 0.3113728665476506, 'subsample': 0.7026635223939202, 'learning_rate': 0.01743143964482159, 'n_estimators': 1707, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:46:09,007] Trial 681 finished with value: 0.11889227704282503 and parameters: {'lambda': 2.467679636845587, 'alpha': 0.06714914520926504, 'colsample_bytree': 0.29936051118699925, 'subsample': 0.6432415875877233, 'learning_rate': 0.015461893185237522, 'n_estimators': 1809, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:46:15,642] Trial 682 finished with value: 0.12349389714850831 and parameters: {'lambda': 6.901234677746283, 'alpha': 0.43553945554442935, 'colsample_bytree': 0.336523600817775,

[I 2023-06-06 08:48:18,043] Trial 700 finished with value: 0.11135144437823685 and parameters: {'lambda': 2.5919672788240655, 'alpha': 1.3504617389872395, 'colsample_bytree': 0.9585164954053109, 'subsample': 0.8090953855411022, 'learning_rate': 0.041546020510207254, 'n_estimators': 1510, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:48:23,813] Trial 701 finished with value: 0.27820553113250385 and parameters: {'lambda': 1.5518331206721032, 'alpha': 3.263535277257497, 'colsample_bytree': 0.39254156469253254, 'subsample': 0.658889715313761, 'learning_rate': 0.0025737013338244727, 'n_estimators': 1576, 'max_depth': 8, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:48:32,052] Trial 702 finished with value: 0.1188949108798525 and parameters: {'lambda': 3.7412589344025946, 'alpha': 1.0204770433125199, 'colsample_bytree': 0.522225524582587,

[I 2023-06-06 08:50:35,181] Trial 720 finished with value: 0.11944591499996421 and parameters: {'lambda': 1.9949409934444973, 'alpha': 1.4895812295397284, 'colsample_bytree': 0.7671237862086469, 'subsample': 0.7148740004882584, 'learning_rate': 0.08984693668778973, 'n_estimators': 1640, 'max_depth': 6, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:50:41,854] Trial 721 finished with value: 0.1202751013629104 and parameters: {'lambda': 1.6194194830183108, 'alpha': 1.383155103947274, 'colsample_bytree': 0.7020244960802532, 'subsample': 0.22992798767119782, 'learning_rate': 0.02019737257116198, 'n_estimators': 1673, 'max_depth': 6, 'min_child_weight': 9, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:50:50,771] Trial 722 finished with value: 0.12870001722772326 and parameters: {'lambda': 2.2272709537086883, 'alpha': 1.231551308833709, 'colsample_bytree': 0.6582725495079038, '

[I 2023-06-06 08:52:21,237] Trial 740 finished with value: 2.345885718350037 and parameters: {'lambda': 1.7628823966751068, 'alpha': 3.8536837644145536, 'colsample_bytree': 0.6726991621300877, 'subsample': 0.6659633598334936, 'learning_rate': 0.001077733619507094, 'n_estimators': 1485, 'max_depth': 4, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:52:24,088] Trial 741 finished with value: 0.13039401964125902 and parameters: {'lambda': 5.602663260987793, 'alpha': 0.0095698225599391, 'colsample_bytree': 0.5840626371785687, 'subsample': 0.6917148344190399, 'learning_rate': 0.018770748232653588, 'n_estimators': 784, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:52:27,836] Trial 742 finished with value: 0.13167468183407585 and parameters: {'lambda': 3.3411361382220672, 'alpha': 8.942996695866723, 'colsample_bytree': 0.6938031017893896, 's

[I 2023-06-06 08:53:44,711] Trial 760 finished with value: 0.1310065008581829 and parameters: {'lambda': 4.300459916280021, 'alpha': 1.5938017927057446, 'colsample_bytree': 0.8246103358073285, 'subsample': 0.5244186173270875, 'learning_rate': 0.08714675172776651, 'n_estimators': 1541, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:53:49,941] Trial 761 finished with value: 0.10043914016397598 and parameters: {'lambda': 1.814474654253618, 'alpha': 0.6558191397337303, 'colsample_bytree': 0.31811984154804374, 'subsample': 0.7581124253955794, 'learning_rate': 0.020666694767968674, 'n_estimators': 1623, 'max_depth': 6, 'min_child_weight': 7, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:53:54,257] Trial 762 finished with value: 0.11717231535555807 and parameters: {'lambda': 1.934512465146813, 'alpha': 2.7960190633191813, 'colsample_bytree': 0.328830181566356, 's

[I 2023-06-06 08:55:08,788] Trial 780 finished with value: 0.1053846851363777 and parameters: {'lambda': 2.8354693094324532, 'alpha': 0.9538243149441583, 'colsample_bytree': 0.8854466410904598, 'subsample': 0.73687071864606, 'learning_rate': 0.015521544447567398, 'n_estimators': 965, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:55:11,891] Trial 781 finished with value: 0.13145695133736682 and parameters: {'lambda': 2.912664892288952, 'alpha': 0.872118225249973, 'colsample_bytree': 0.8606334781206118, 'subsample': 0.7358088300001894, 'learning_rate': 0.015375699009119849, 'n_estimators': 943, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:55:14,711] Trial 782 finished with value: 0.14292879718406984 and parameters: {'lambda': 3.127897331102422, 'alpha': 6.40775204744096, 'colsample_bytree': 0.8535955745087515, 'subs

[I 2023-06-06 08:56:23,505] Trial 800 finished with value: 0.11789312171577783 and parameters: {'lambda': 2.9571368127435993, 'alpha': 0.6699441019724719, 'colsample_bytree': 0.9334831110077586, 'subsample': 0.7580419557166771, 'learning_rate': 0.027663721433617193, 'n_estimators': 1056, 'max_depth': 3, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:56:27,015] Trial 801 finished with value: 0.13010901036334604 and parameters: {'lambda': 1.581446661398123, 'alpha': 5.245051105481689, 'colsample_bytree': 0.9404977457792215, 'subsample': 0.7458431890747128, 'learning_rate': 0.022829059345206466, 'n_estimators': 972, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:56:29,503] Trial 802 finished with value: 0.1538425616943315 and parameters: {'lambda': 2.6954517465645997, 'alpha': 9.255381341516557, 'colsample_bytree': 0.9557433173872737, 

[I 2023-06-06 08:57:42,889] Trial 820 finished with value: 0.12986521362184894 and parameters: {'lambda': 0.5527214223572383, 'alpha': 1.2626992601763376, 'colsample_bytree': 0.9990146217665805, 'subsample': 0.6591531173327908, 'learning_rate': 0.031289555187320066, 'n_estimators': 2026, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:57:53,580] Trial 821 finished with value: 0.12400561707747045 and parameters: {'lambda': 0.9012354638426585, 'alpha': 0.3821793884240299, 'colsample_bytree': 0.8325634580339796, 'subsample': 0.61071473289133, 'learning_rate': 0.025256555626461553, 'n_estimators': 2226, 'max_depth': 9, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:58:01,769] Trial 822 finished with value: 0.14531764898064156 and parameters: {'lambda': 1.327459125062348, 'alpha': 0.6638385047435342, 'colsample_bytree': 0.8075652868926829, 

[I 2023-06-06 08:59:54,326] Trial 840 finished with value: 0.15603247914828877 and parameters: {'lambda': 3.040030050783623, 'alpha': 0.6183320996131387, 'colsample_bytree': 0.9341579258914131, 'subsample': 0.290825342013438, 'learning_rate': 0.051428567795271225, 'n_estimators': 1882, 'max_depth': 4, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 08:59:59,872] Trial 841 finished with value: 0.14755343943743635 and parameters: {'lambda': 1.2331249371482098, 'alpha': 9.052555126032422, 'colsample_bytree': 0.9169782830544613, 'subsample': 0.7344042535174985, 'learning_rate': 0.09463246000784097, 'n_estimators': 2172, 'max_depth': 4, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:00:02,127] Trial 842 finished with value: 0.13372268355594527 and parameters: {'lambda': 1.827978638123068, 'alpha': 1.005902891208609, 'colsample_bytree': 0.8796131369151948, 'su

[I 2023-06-06 09:01:43,713] Trial 860 finished with value: 0.12008916139647394 and parameters: {'lambda': 3.300488913809845, 'alpha': 1.4884836735700562, 'colsample_bytree': 0.9784851031984495, 'subsample': 0.2288522520440851, 'learning_rate': 0.07129934189970624, 'n_estimators': 1914, 'max_depth': 6, 'min_child_weight': 9, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:01:48,926] Trial 861 finished with value: 0.13048563238713828 and parameters: {'lambda': 2.444351358811627, 'alpha': 5.593905213137605, 'colsample_bytree': 0.47886865342361806, 'subsample': 0.5536715871086004, 'learning_rate': 0.06678701935706584, 'n_estimators': 2344, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:01:54,013] Trial 862 finished with value: 0.14298096170926683 and parameters: {'lambda': 1.7619099174664825, 'alpha': 1.199441312644351, 'colsample_bytree': 0.8595314574874309, 's

[I 2023-06-06 09:03:07,521] Trial 880 finished with value: 0.13405242237221687 and parameters: {'lambda': 2.841330445831444, 'alpha': 0.9962912812044689, 'colsample_bytree': 0.3087827216512449, 'subsample': 0.37251700322489156, 'learning_rate': 0.025188499956197263, 'n_estimators': 1846, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:03:13,853] Trial 881 finished with value: 0.11501841562419132 and parameters: {'lambda': 3.363165659860298, 'alpha': 1.5475915647693308, 'colsample_bytree': 0.8741108890209884, 'subsample': 0.7657419371929568, 'learning_rate': 0.03214911728471133, 'n_estimators': 2132, 'max_depth': 6, 'min_child_weight': 2, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:03:16,121] Trial 882 finished with value: 0.11924033623116467 and parameters: {'lambda': 9.591518445137504, 'alpha': 1.3201918252332454, 'colsample_bytree': 0.4506894695583841,

[I 2023-06-06 09:04:12,142] Trial 900 finished with value: 0.12375859230412514 and parameters: {'lambda': 1.2554182725692247, 'alpha': 0.6562245301416509, 'colsample_bytree': 0.8310361874430192, 'subsample': 0.4485801555949963, 'learning_rate': 0.04516910551462326, 'n_estimators': 745, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:04:14,505] Trial 901 finished with value: 0.14348385730513727 and parameters: {'lambda': 1.04942661618703, 'alpha': 6.17558491840124, 'colsample_bytree': 0.7658845312859902, 'subsample': 0.7504516050956441, 'learning_rate': 0.07387769340536078, 'n_estimators': 962, 'max_depth': 6, 'min_child_weight': 8, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:04:19,293] Trial 902 finished with value: 0.11698028798630132 and parameters: {'lambda': 3.0671345999293758, 'alpha': 5.849859643839812, 'colsample_bytree': 0.29999093055914233, 'sub

[I 2023-06-06 09:05:22,464] Trial 920 finished with value: 0.11951212343413994 and parameters: {'lambda': 1.1454992955868548, 'alpha': 0.6551125031619827, 'colsample_bytree': 0.9459801253329516, 'subsample': 0.34149844484134756, 'learning_rate': 0.08416169037591989, 'n_estimators': 1018, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:05:28,290] Trial 921 finished with value: 0.11668940378741083 and parameters: {'lambda': 9.47253281624242, 'alpha': 1.0068867777481922, 'colsample_bytree': 0.458919205147025, 'subsample': 0.38648342789123113, 'learning_rate': 0.03183808731295597, 'n_estimators': 2152, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:05:33,224] Trial 922 finished with value: 0.13728937813723052 and parameters: {'lambda': 3.282746851128523, 'alpha': 2.204496909340925, 'colsample_bytree': 0.7602481606777884, '

[I 2023-06-06 09:06:41,198] Trial 940 finished with value: 0.1260186890880539 and parameters: {'lambda': 1.4134276373290333, 'alpha': 1.5572406384633097, 'colsample_bytree': 0.2613760203126053, 'subsample': 0.5630882354300253, 'learning_rate': 0.07509516955420019, 'n_estimators': 1888, 'max_depth': 6, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:06:42,001] Trial 941 finished with value: 0.14340988489761958 and parameters: {'lambda': 3.4052382676833446, 'alpha': 5.199467358436928, 'colsample_bytree': 0.30112650968135674, 'subsample': 0.5852550079066619, 'learning_rate': 0.05135512543881216, 'n_estimators': 183, 'max_depth': 10, 'min_child_weight': 1, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:06:46,771] Trial 942 finished with value: 0.12891210356617733 and parameters: {'lambda': 2.824051199299805, 'alpha': 2.520485247262162, 'colsample_bytree': 0.8666836823260651, 's

[I 2023-06-06 09:07:59,426] Trial 960 finished with value: 0.12595104249373462 and parameters: {'lambda': 5.105348135167411, 'alpha': 9.187319215140963, 'colsample_bytree': 0.3160875829109166, 'subsample': 0.7401945151694266, 'learning_rate': 0.06746443479535177, 'n_estimators': 2116, 'max_depth': 6, 'min_child_weight': 10, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:08:04,687] Trial 961 finished with value: 0.12747634367857952 and parameters: {'lambda': 2.6529286444748874, 'alpha': 0.9577665105316276, 'colsample_bytree': 0.238036900447376, 'subsample': 0.8055874870530412, 'learning_rate': 0.08598386417015848, 'n_estimators': 2048, 'max_depth': 5, 'min_child_weight': 5, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:08:08,859] Trial 962 finished with value: 0.11654195782895242 and parameters: {'lambda': 4.063458752641315, 'alpha': 1.803974473278013, 'colsample_bytree': 0.3782130016914927, 'su

[I 2023-06-06 09:09:21,759] Trial 980 finished with value: 0.14296306042259949 and parameters: {'lambda': 2.0106902365070383, 'alpha': 1.1414471817085006, 'colsample_bytree': 0.5529730980960956, 'subsample': 0.8587219834766833, 'learning_rate': 0.0333226847498769, 'n_estimators': 1058, 'max_depth': 3, 'min_child_weight': 3, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:09:26,342] Trial 981 finished with value: 0.1299011692135734 and parameters: {'lambda': 2.970463134224524, 'alpha': 2.6133868281943626, 'colsample_bytree': 0.36993416764433146, 'subsample': 0.7768779889606006, 'learning_rate': 0.028394779129190306, 'n_estimators': 1695, 'max_depth': 6, 'min_child_weight': 4, 'num_parallel_tree': 1}. Best is trial 173 with value: 0.09909467168372289.
[I 2023-06-06 09:09:27,918] Trial 982 finished with value: 0.1277912448878613 and parameters: {'lambda': 4.8012841808924165, 'alpha': 0.7080502884954265, 'colsample_bytree': 0.2613811843823692, '

Number of finished trials: 1000
Best trial: {'lambda': 1.32754289013568, 'alpha': 1.580513088436275, 'colsample_bytree': 0.34112979113700337, 'subsample': 0.49829211790557504, 'learning_rate': 0.04245317476432714, 'n_estimators': 1145, 'max_depth': 6, 'min_child_weight': 6, 'num_parallel_tree': 1}


In [26]:
X,y = shuffle(X,y, random_state=42)
X = X.reset_index(drop=True)
y = y.reset_index(drop=True)

In [27]:
def rmse_cv(model,X,y):
    rmse = np.sqrt(-cross_val_score(model, X, y, scoring="neg_mean_squared_error", cv=6))
    return rmse

In [28]:
params_xgb= {'lambda': 1.32754289013568, 
             'alpha': 1.580513088436275, 
             'colsample_bytree': 0.34112979113700337, 
             'subsample': 0.49829211790557504, 
             'learning_rate': 0.04245317476432714, 
             'n_estimators': 1145,
             'max_depth': 6, 
             'min_child_weight': 6, 
             'num_parallel_tree': 1}

In [29]:
xgb_model = xgb.XGBRegressor(**params_xgb)

In [30]:
print(rmse_cv(xgb_model,X,y).mean())

0.12540821139707464


In [31]:
submit = test[['Id']]
submit = submit.reset_index(drop=True)

In [32]:
xgb_model.fit(X,y)

XGBRegressor(alpha=1.580513088436275, base_score=None, booster=None,
             callbacks=None, colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=0.34112979113700337, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, lambda=1.32754289013568,
             learning_rate=0.04245317476432714, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=6, max_leaves=None,
             min_child_weight=6, missing=nan, monotone_constraints=None,
             n_estimators=1145, n_jobs=None, num_parallel_tree=1, ...)

In [33]:
submit_predict = xgb_model.predict(test)
submit_predict = np.exp(submit_predict)

In [34]:
submit['SalePrice'] = submit_predict

In [35]:
submit.to_csv('submit_xgb_optuna.csv', index=False)

Reducing feature skewness actually slightly increased the Kaggle score. 